mcs/class/Mono.Posix/Mono.Unix.Native/Syscall.cs

```diff
@@ -727,6 +727,194 @@ public enum MremapFlags : ulong {
 		MREMAP_MAYMOVE = 0x1,
 	}
 
+	[Map]
+	[CLSCompliant (false)]
+	public enum SockType : int {
+		SOCK_STREAM    =  1, // Byte-stream socket
+		SOCK_DGRAM     =  2, // Datagram socket
+		SOCK_RAW       =  3, // Raw protocol interface (linux specific)
+		SOCK_RDM       =  4, // Reliably-delivered messages (linux specific)
+		SOCK_SEQPACKET =  5, // Sequenced-packet socket
+		SOCK_DCCP      =  6, // Datagram Congestion Control Protocol (linux specific)
+		SOCK_PACKET    = 10, // Linux specific
+	}
+
+	[Map][Flags]
+	[CLSCompliant (false)]
+	public enum SockFlags : int {
```

Comment:
This won't work without a change to create-native-map because SocketFlags needs the [Flags] attribute but SocketType must not have it. (The mapping code for flags is different than for "real" enumerations.)
OpenFlags has the same problem but it seems to work by accident because O_RDONLY is 0 and O_WRONLY and O_RDWR are powers of two.
Anyway, I think it is cleaner two split the type and the flags into two values.